.github/workflows/golangci-lint.yml

```diff
@@ -2,8 +2,14 @@ name: golangci-lint
 on:
   pull_request:
     types: [opened, edited, synchronize, reopened]
+permissions:
+  contents: read
+
 jobs:
   golangci:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
+      pull-requests: read  # for golangci/golangci-lint-action to fetch pull requests
```

Comment:
Q: Does the golangci-lint action really fetch PRs?

Reply:
>   # Optional: allow read access to pull request. Use with `only-new-issues` option.

xref: https://github.com/golangci/golangci-lint-action

So in our case I think it is not necessary.